### PR TITLE
fix: 🐛 Fix search text not emptied on close

### PR DESCRIPTION
**Pull request by chef-ryan**
---

## PR-Codex overview
This PR focuses on improving the dismissal handling of the `AddGaugeModal` component by introducing a `dismissHandler` function. This function resets the search text and calls the `onDismiss` callback, ensuring a consistent state when the modal is closed.

### Detailed summary
- Added `dismissHandler` function using `useCallback` to reset search text and handle modal dismissal.
- Updated the `onDismiss` prop in `<ModalV2>` to use `dismissHandler` instead of directly calling `onDismiss`.
- Changed the `onClick` prop of the close button to use `dismissHandler`.
- Updated the `onClick` prop of `<BottomAction>` to use `dismissHandler`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/views/GaugesVoting/components/Table/AddGauge/AddGaugeModal.tsx [hunk 1]
```diff
@@ -87,7 +87,15 @@ export const AddGaugeModal = ({ isOpen, onDismiss, selectRows, onGaugeAdd }) =>
                     {t('Search and add the gauge you want to vote')}
                   </Text>
                 </AutoColumn>
-                <Button variant="text" onClick={onDismiss} px={0} height="fit-content">
+                <Button
+                  variant="text"
+                  onClick={() => {
```
[the remark this answers is not in the document]
Comment (ChefJerry) [reply apps/web/src/views/GaugesVoting/components/Table/AddGauge/AddGaugeModal.tsx:92]: how about add a effect to handle this case

```
useEffect(() => {
  if (!isOpen) setSearchText('')
  
  return () => setSearchText('')
}, [])
```

[the remark this answers is not in the document]
Comment (chef-ryan) [reply apps/web/src/views/GaugesVoting/components/Table/AddGauge/AddGaugeModal.tsx]: Just add a wrapper function ...

[the remark this answers is not in the document]
Comment (ChefJerry) [reply apps/web/src/views/GaugesVoting/components/Table/AddGauge/AddGaugeModal.tsx]: ah, it is used for close button

File: apps/web/src/views/GaugesVoting/components/Table/AddGauge/AddGaugeModal.tsx [hunk 2]
```diff
@@ -42,6 +43,10 @@ export const AddGaugeModal = ({ isOpen, onDismiss, selectRows, onGaugeAdd }) =>
   const totalGaugesWeight = useGaugesTotalWeight()
   const { data: gauges } = useGauges()
   const { filterGauges, setSearchText, filter, setFilter, setSort } = useGaugesFilter(gauges)
+  const dismissHandler = useCallback(() => {
+    setSearchText('')
+    onDismiss()
+  }, [])
```
Comment (ChefJerry) [inline apps/web/src/views/GaugesVoting/components/Table/AddGauge/AddGaugeModal.tsx:49]: `onDismiss` should in deps

Comment (chef-ryan) [reply apps/web/src/views/GaugesVoting/components/Table/AddGauge/AddGaugeModal.tsx]: added. onDismiss is required so no undefined check here.

**Review by ChefMomota — APPROVED** [review]

**Pull request merged**